src/logic/currencyValues/store/reducer/currencyValues.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+import { Map } from 'immutable'
+import { handleActions, type ActionType } from 'redux-actions'
+import { SET_CURRENCY_BALANCES } from '../actions/setCurrencyBalances'
+import type { State } from '~/logic/tokens/store/reducer/tokens'
+import { SET_CURRENT_CURRENCY } from '~/logic/currencyValues/store/actions/setCurrencySelected'
+
+export const CURRENCY_VALUES_KEY = 'currencyValues'
+
+export default handleActions<State, *>(
+  {
+    [SET_CURRENCY_BALANCES]: (state: State, action: ActionType<Function>): State => {
+      const { currencyBalances } = action.payload
+
+      const newState = state.withMutations((map) => {
+        map.set('currencyBalances', currencyBalances)
```

Comment:
doesn't need `withMutations`
https://stackoverflow.com/questions/28510753/when-should-i-use-withmutations-on-a-map-in-immutable-js/29381227

Reply:
Done